caffe2/CMakeLists.txt

```diff
@@ -1195,6 +1195,18 @@ if(USE_CUDA)
 
 endif()
 
+# ---[ Metal(OSX) modification
+if(APPLE AND USE_PYTORCH_METAL)
```

Comment:
nit: can we make 'USE_PYTORCH_METAL' only enabled when 'APPLE' is true?

Reply:
`USE_PYTORCH_METAL` is also available in Linux. In that case, only one cpp file under `aten/native/metal` will be compiled. The reason is that users can still do `optimize_for_mobile` on Linux machines to export models for metal.